File: backend/src/main/java/com/medvault/backend/service/MedicalRecordService.java
package com.medvault.backend.service;

import com.medvault.backend.dto.request.medical.*;
import java.util.List;

public interface MedicalRecordService {
    // Allergy
    AllergyDto addAllergy(String email, AllergyDto dto);
    List<AllergyDto> getAllergies(String email);
    void deleteAllergy(String email, Long id);

    // ChronicDisease
    ChronicDiseaseDto addDisease(String email, ChronicDiseaseDto dto);
    List<ChronicDiseaseDto> getDiseases(String email);
    void deleteDisease(String email, Long id);

    // Medication
    MedicationDto addMedication(String email, MedicationDto dto);
    List<MedicationDto> getMedications(String email);
    void deleteMedication(String email, Long id);

    // Surgery
    SurgeryDto addSurgery(String email, SurgeryDto dto);
    List<SurgeryDto> getSurgeries(String email);
    void deleteSurgery(String email, Long id);

    // Immunization
    ImmunizationDto addImmunization(String email, ImmunizationDto dto);
    List<ImmunizationDto> getImmunizations(String email);
    void deleteImmunization(String email, Long id);

    // FamilyHistory
    FamilyHistoryDto addFamilyHistory(String email, FamilyHistoryDto dto);
    List<FamilyHistoryDto> getFamilyHistories(String email);
    void deleteFamilyHistory(String email, Long id);
}

File: frontend/src/services/medicalRecordService.js
import api from './api';

export const medicalRecordService = {
  // Allergies
  getAllergies: async () => (await api.get('/patient/medical-records/allergies')).data,
  addAllergy: async (data) => (await api.post('/patient/medical-records/allergies', data)).data,
  deleteAllergy: async (id) => (await api.delete(`/patient/medical-records/allergies/${id}`)).data,

  // Diseases
  getDiseases: async () => (await api.get('/patient/medical-records/diseases')).data,
  addDisease: async (data) => (await api.post('/patient/medical-records/diseases', data)).data,
  deleteDisease: async (id) => (await api.delete(`/patient/medical-records/diseases/${id}`)).data,

  // Medications
  getMedications: async () => (await api.get('/patient/medical-records/medications')).data,
  addMedication: async (data) => (await api.post('/patient/medical-records/medications', data)).data,
  deleteMedication: async (id) => (await api.delete(`/patient/medical-records/medications/${id}`)).data,

  // Surgeries
  getSurgeries: async () => (await api.get('/patient/medical-records/surgeries')).data,
  addSurgery: async (data) => (await api.post('/patient/medical-records/surgeries', data)).data,
  deleteSurgery: async (id) => (await api.delete(`/patient/medical-records/surgeries/${id}`)).data,

  // Immunizations
  getImmunizations: async () => (await api.get('/patient/medical-records/immunizations')).data,
  addImmunization: async (data) => (await api.post('/patient/medical-records/immunizations', data)).data,
  deleteImmunization: async (id) => (await api.delete(`/patient/medical-records/immunizations/${id}`)).data,

  // Family Histories
  getFamilyHistories: async () => (await api.get('/patient/medical-records/family-histories')).data,
  addFamilyHistory: async (data) => (await api.post('/patient/medical-records/family-histories', data)).data,
  deleteFamilyHistory: async (id) => (await api.delete(`/patient/medical-records/family-histories/${id}`)).data,
};
